### PR TITLE
Sphinx include readme

### PR DESCRIPTION
**Pull request by dbkeitel**
Uses [m2r2](https://pypi.org/project/m2r2/) to directly include the `README.md` into the sphinx html pages. Notes:

- most tutorials still recommend the original `m2r` but it seems to be abandoned
- md->rst involves conforming to rst's very picky standards, which include issuing warnings for multiple external links with the displayed text ("Duplicate explicit target name"). I didn't find a simple workaround, so I just modified the README to only use unique link texts.
- `readme.rst` doesn't have a heading on purpose, because that would lead to duplication in the TOC.

Also:

- [x] add `Pillow` dependency since, according to https://sphinx-gallery.github.io/stable/index.html#install-via-pip:
> Sphinx-Gallery will not manage its dependencies when installing
- [x] followup `.gitignore` fix from #247

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -5,10 +5,14 @@ NOTE the html documentation is work in progress,
 likely contains many formatting issues,
 and is not guaranteed to be complete.
 
+The main repository for PYFstat can be found at
```
Comment (Rodrigo-Tenorio) [inline docs/source/index.rst:8]:
```suggestion
The main repository for PyFstat can be found at
```

File: docs/source/modules.rst [hunk 2]
```diff
@@ -2,8 +2,8 @@
    sphinx-apidoc -f -o source ../ ../versioneer.py ../setup.py
    (everything after the first "../" are exclude patterns)
 
-PyFstat
-=======
+PyFstat module documentation
+============================
```
Comment (Rodrigo-Tenorio) [inline docs/source/modules.rst:6]:
```suggestion
===========================
```

**Review by Rodrigo-Tenorio — CHANGES_REQUESTED** [review]
There are two typos/syntax errors (not sure where to draw the line anymore). 

The rest looks good to me!

File: docs/source/modules.rst (hunk 2 above)
Comment (dbkeitel) [reply docs/source/modules.rst:6]: Looks correct to me as is...?

**Comment by dbkeitel** [comment]
- [x] Improved the intro paragraphs of the various pages.
- [x] Learned how to do both external and internal links. (Not how to link to sections inside a single m2r2-converted file though, but that'd probably be asking too much.)

**Review by Rodrigo-Tenorio — APPROVED** [review]

**Pull request merged**